fvm/evm/emulator/emulator_test.go

```diff
@@ -0,0 +1,371 @@
+							testContract.MakeStoreCallData(t, num),
+							1_000_000,
+							big.NewInt(0), // this should be zero because the contract doesn't have receiver
+						),
+					)
+					require.NoError(t, err)
+					require.GreaterOrEqual(t, res.GasConsumed, uint64(40_000))
+				})
+			})
+
+			RunWithNewEmulator(t, db, func(env *emulator.Emulator) {
+				RunWithNewBlockView(t, env, func(blk types.BlockView) {
+					res, err := blk.DirectCall(
+						types.NewContractCall(
+							testAccount,
+							contractAddr,
+							testContract.MakeRetrieveCallData(t),
+							1_000_000,
+							big.NewInt(0), // this should be zero because the contract doesn't have receiver
+						),
+					)
+					require.NoError(t, err)
+
+					ret := new(big.Int).SetBytes(res.ReturnedValue)
+					require.Equal(t, num, ret)
+					require.GreaterOrEqual(t, res.GasConsumed, uint64(23_000))
+				})
+			})
+
+			RunWithNewEmulator(t, db, func(env *emulator.Emulator) {
+				RunWithNewBlockView(t, env, func(blk types.BlockView) {
+					res, err := blk.DirectCall(
+						types.NewContractCall(
+							testAccount,
+							contractAddr,
+							testContract.MakeBlockNumberCallData(t),
+							1_000_000,
+							big.NewInt(0), // this should be zero because the contract doesn't have receiver
+						),
+					)
+					require.NoError(t, err)
+
+					ret := new(big.Int).SetBytes(res.ReturnedValue)
+					require.Equal(t, blockNumber, ret)
+				})
+			})
+
+		})
+
+		t.Run("test sending transactions (happy case)", func(t *testing.T) {
+			account := testutils.GetTestEOAAccount(t, testutils.EOATestAccount1KeyHex)
+			fAddr := account.Address()
+			RunWithNewEmulator(t, db, func(env *emulator.Emulator) {
+				RunWithNewBlockView(t, env, func(blk types.BlockView) {
+					_, err := blk.DirectCall(types.NewDepositCall(fAddr, amount))
+					require.NoError(t, err)
+				})
+			})
+
+			RunWithNewEmulator(t, db, func(env *emulator.Emulator) {
+				ctx := types.NewDefaultBlockContext(blockNumber.Uint64())
+				ctx.GasFeeCollector = types.NewAddressFromString("coinbase")
+				coinbaseOrgBalance := gethCommon.Big1
+				// small amount of money to create account
+				RunWithNewBlockView(t, env, func(blk types.BlockView) {
+					_, err := blk.DirectCall(types.NewDepositCall(ctx.GasFeeCollector, coinbaseOrgBalance))
+					require.NoError(t, err)
+				})
+
+				blk, err := env.NewBlockView(ctx)
+				require.NoError(t, err)
+				tx := account.PrepareAndSignTx(
+					t,
+					testAccount.ToCommon(), // to
+					nil,                    // data
+					big.NewInt(1000),       // amount
+					gethParams.TxGas,       // gas limit
+					gethCommon.Big1,        // gas fee
+
+				)
+				_, err = blk.RunTransaction(tx)
+				require.NoError(t, err)
+
+				// check the balance of coinbase
+				RunWithNewReadOnlyBlockView(t, env, func(blk2 types.ReadOnlyBlockView) {
+					bal, err := blk2.BalanceOf(ctx.GasFeeCollector)
+					require.NoError(t, err)
+					require.Greater(t, bal.Uint64(), coinbaseOrgBalance.Uint64())
```

Comment:
Could we be precise:
pseudo code
```
gethParams.TxGas * gethCommon.Big1 + gethCommon.Big1
```

Reply:
updated at [c848b92](https://github.com/onflow/flow-go/pull/4927/commits/c848b9212ca242dd68b095c4a3cd63058af8c377)